### ISSUE TITLE
n1850.ne30_tn14.hybrid_fatessp_cplhist.202401031

### ISSUE THREAD

**Issue by mvertens** [NorESMhub/noresm3_dev_simulations#52]
**Purpose:**
Run for 2 years generating auxiliary mediatory history output needed to spin up ocean and land.
Duplicate of n1850.ne30_tn14.hybrid_fatessp.202401031 but with a fix for DMS in CAM.  The goal is to run 2 years.

___
**Description:**
- fix for DMS being set to zero inside CAM despite non-zero values coming from the ocean (see https://github.com/NorESMhub/CAM/pull/174)
- use of FATES-SP and spun up initial conditions for FATES-SP using offline spinup
- NOTE: a memory leak is present in the run and hits agains the node `limit` when running past 11 years.
See https://github.com/NorESMhub/NorESM/issues/550

___
**Description:**

- **create_newcase commnad**
```
 ./create_newcase --case <user_case_path>/n1850.ne30_tn14.hybrid_fatessp.202401031 \\
 --compset 1850_CAM%DEV%LT%NORESM%CAMoslo_CLM60%FATES-SP_CICE_BLOM%ECO_MOSART_DGLC%NOEVOLVE_SWAV_SESP  \\
 --res ne30pg3_tn14 \\
 --project nn9039k --run-unsupported --mach betzy 
```

- **user_nl_cpl**
```
 histaux_atm2med_file1_enabled = .true.
 histaux_atm2med_file2_enabled = .true.
 histaux_atm2med_file3_enabled = .true.
 histaux_atm2med_file4_enabled = .true.
 histaux_atm2med_file5_enabled = .true.
 histaux_rof2med_file1_enabled = .true.

```
- **user_nl_cam**
```
 use_aerocom = .true.
 interpolate_nlat   = 192
 interpolate_nlon   = 288
 interpolate_output = .true.
 history_aerosol = .true. 
 zmconv_c0_lnd  =  0.0075D0
 zmconv_c0_ocn  =  0.0300D0
 zmconv_ke =  5.0E-6
 zmconv_ke_lnd =  1.0E-5
 clim_modal_aero_top_press =  1.D-4
 bndtvg = '/cluster/shared/noresm/inputdata/atm/cam/ggas/noaamisc.r8.nc'
```
- **user_nl_blom**
`ICFILE = "/cluster/shared/noresm/inputdata/restart/NOIIAJRAOC20TR_TL319_tn14_ppm_20240816/rest/1775-01-01-00000/NOIIAJRAOC20TR_TL319_tn14_ppm_20240816.blom.r.1775-01-01-00000.nc"`

- **user_nl_cice**
`ice_ic = "/cluster/shared/noresm/inputdata/restart/NOIIAJRAOC20TR_TL319_tn14_ppm_20240816/rest/1775-01-01-00000/NOIIAJRAOC20TR_TL319_tn14_ppm_20240816.cice.r.1775-01-01-00000.nc"`

- **user_nl_clm**
Add the following to user_nl_clm
```
finidat='/cluster/shared/noresm/inputdata/lnd/clm2/initdata_esmf/ctsm5.3/testing/1850_fates_spinup_new_dust.clm2.r.0081-01-01-00000.nc'
z0param_method='ZengWang2007'
```
- **xmlchange commands**
```
./xmlchange NTASKS=1920
./xmlchange NTASKS_OCN=256
./xmlchange ROOTPE=0
./xmlchange ROOTPE_OCN=1920
./xmlchange BLOM_VCOORD=cntiso_hybrid,BLOM_TURBULENT_CLOSURE=
./xmlchange STOP_OPTION=nyears
./xmlchange STOP_N=7
./xmlchange REST_N=1
./xmlchange REST_OPTION=nyears
./xmlchange RESUBMIT=3
./xmlchange --subgroup case.run JOB_WALLCLOCK_TIME=48:00:00
./xmlchange --subgroup case.st_archive JOB_WALLCLOCK_TIME=03:00:00
```
___
**Compset long name**
`1850_CAM%DEV%LT%NORESM%CAMoslo_CLM60%FATES-SP_CICE_BLOM%ECO_MOSART_DGLC%NOEVOLVE_SWAV_SESP`

___
**Model grid long name**
`a%ne30np4.pg3_l%ne30pg4.pg3_oi%tnx1v4_r%r05_w%null_z%null_g%gris4_m%tnx1v4`

___
**Case directory:**
- Locally on betzy: 
   `<user_pathname>/n1850.ne30_tn14.hybrid_fatessp.202401031`
- On github: 
 
___
**Code version on github:**
- https://github.com/NorESMhub/NorESM/releases/tag/noresm2_5_alpha07
- CAM needs to be updated - modify Externals.cfg to have the following change:
```
[cam]
protocol = git
tag = noresm2_5_015_cam6_3_158.  
branch = noresm_develop
repo_url = https://github.com/NorESMhub/CAM
local_path = components/cam
externals = Externals_CAM.cfg
required = True
```

___
**Diagnostics:**

___
**Output:**
- Raw output: ` /nird/datalake/NS9560K/noresm3/cases/n1850.ne30_tn14.hybrid_fatessp.20241031 `
___
**Contacts:**
@mvertens, @monsieuralok 
___
**Extra details:**

**Comment by tjiputra** [comment]
The above Externals.cfg modification didn't work. I have to remove 

"branch = noresm_develop"

Is this correct, @mvertens ?

Also, this is a hybrid start? Is there a reference case name?

**Comment by mvertens** [comment]
@tjiputra - this is a startup run. All the xml changes are in the directions. Sorry for the mistaken change for the code. I did test that this worked for a 10 day run. Please let me know if you have any problems or questions.

**Comment by tjiputra** [comment]
finidat='/cluster/shared/noresm/inputdata/lnd/clm2/initdata_esmf/ctsm5.3/testing/1850_fates_spinup_new_dust.clm2.r.0081-01-01-00000.nc'

With this in user_nl_clm, the model won't compile, says it has to be hybrid or branch.

**Comment by mvertens** [comment]
This is very odd. Let me try to duplicate this problem. I simply duplicated the directions for the last fates-sp simulation.
Sorry you are running into this.

**Comment by mvertens** [comment]
The error message says:
```
err=Warning : CLM build-namelist::CLMBuildNamelist::setup_logic_initial_conditions() : setting finidat (either explicitly in your user_nl_clm or by doing a hybrid or branch RUN_TYPE)
 is incomptable with using a cold start (by setting CLM_FORCE_COLDSTART=on).
 -- Add -ignore_warnings option to CLM_BLDNML_OPTS to ignore this warning
```
You need to issue the following in your $CASEROOT:
`./xmlchange --append  CLM_BLDNML_OPTS="-ignore_warnings"`
I'm not sure how @monsieuralok ran the previous FATES case  - #44 - without this. 
I have updated the directions above as well as in #44 as well so we can keep track of this.

**Comment by rosiealice** [comment]
Is the problem potentially related to this? 

https://github.com/NGEET/fates/issues/1234

**Comment by mvertens** [comment]
> Is the problem potentially related to this?
> 
> [NGEET/fates#1234](https://github.com/NGEET/fates/issues/1234)
> 
> (doesn't really answer the question of how we ran the last simulation...)
I don't think that is the problem. The error statement itself does not make sense to me since CLM_FORCE_COLDSTART=off.
The only way I see that @monsieuralok must have been able to run is to set 
`./xmlchange --append  CLM_BLDNML_OPTS="-ignore_warnings"`
@monsieuralok - can you confirm this?

**Comment by monsieuralok** [comment]
@mvertens @rosiealice I have not given intial file for CLM so removed "finidat" as mentioned in meeting.

**Comment by mvertens** [comment]
@tjiputra - it turns out that changes were introduced in cmeps since the first CESM spin up tests were run cause problems in the generation of some of the auxiliary files. I realized this when I tried to use your data to read in forcings using DATM. The following new settings will need to be added to `user_nl_cpl`
  ```
 histaux_atm2med_file5_ntperfile = 1
 histaux_atm2med_file5_history_n = 1
 histaux_atm2med_file5_history_option = 'ndays'
 histaux_rof2med_file1_ntperfile = 1
 histaux_rof2med_file1_history_n = 1
 histaux_rof2med_file1_history_option = 'ndays'
```
And there will be a lot fewer files.
Would you be willing to delete the generated cpl hist auxiliary files and run the two years again and move the data to nird?
I've validated that with these settings correct auxiliary files are produced and datm can read them in and force clm. So I can duplicate that with the new data as well and also verify that this works with DATM/BLOM.

**Comment by tjiputra** [comment]
@mvertens I can delete and redo the run. Could you check how we can generate the SMB fields that the icesheet needs in this run as well?
https://github.com/NorESMhub/noresm3_dev_simulations/discussions/50

**Comment by mvertens** [comment]
@tjiputra - good point. I'll test this and add to the directions. Thanks for thinking of this!

**Comment by tjiputra** [comment]
@mvertens i will go a head and retest this and let you know when they are ready.

Could you update the xmlchange commands to the following:

./xmlchange NTASKS=1920
./xmlchange NTASKS_OCN=256
./xmlchange ROOTPE=0
./xmlchange ROOTPE_OCN=1920
./xmlchange BLOM_VCOORD=cntiso_hybrid,BLOM_TURBULENT_CLOSURE=
./xmlchange STOP_OPTION=nyears
./xmlchange STOP_N=2
./xmlchange --subgroup case.run JOB_WALLCLOCK_TIME=12:00:00
./xmlchange --subgroup case.st_archive JOB_WALLCLOCK_TIME=03:00:00
./xmlchange --append  CLM_BLDNML_OPTS="-ignore_warnings"
